### PR TITLE
[writer] Emit metrics on length of ids and number of policies

### PR DESCRIPTION
**Pull request by jeromefroe**
This PR adds histograms for the length of ID's and the number of policies in each metric seen so far.

cc @xichen2020

### CODE REVIEW

File: backend/msgpack/writer.go [hunk 1]
```diff
@@ -91,6 +91,8 @@ func (w *writer) Write(
 	mu unaggregated.MetricUnion,
 	pl policy.PoliciesList,
 ) error {
+	w.metrics.idLengths.RecordValue(float64(len(mu.ID)))
```
Comment (xichen2020) [inline backend/msgpack/writer.go]: This is called on the critical path of writing every metric, are we worried about the CPU overhead of doing a binary search and recording it?

Comment (jeromefroe) [reply backend/msgpack/writer.go]: I couldn't find any tally benchmarks for recording a histogram so I made a quick one myself, recording a value took on average 35ns. So I think a fair estimate for the time these two function calls take in total would be 100ns. Even if a collector is processing 100,000 metrics per second, that's only an additional 10ms of CPU time. With that being said, just by the law of the large numbers, even if we only record metrics for a fraction of the metrics we see, we should still get a very close approximation to the true distribution of id lengths and number of policies, so I'll add a dice here so we can control what fraction of metrics we record these values for. I think these would be value generally so I'm hesitant to not record them at all.

File: backend/msgpack/writer.go [hunk 2]
```diff
@@ -91,6 +96,10 @@ func (w *writer) Write(
 	mu unaggregated.MetricUnion,
 	pl policy.PoliciesList,
 ) error {
+	if w.rand.Float64() < w.samplingRate {
```
Comment (jeromefroe) [inline backend/msgpack/writer.go]: Some benchmarks I ran show that a call to `Float64` takes on the order of 5ns when using a lock `Rand`: https://github.com/jeromefroe/golang_benchmarks#rand

File: backend/msgpack/writer.go [hunk 3]
```diff
@@ -76,6 +79,8 @@ func newInstanceWriter(instance placement.Instance, opts ServerOptions) instance
 	w := &writer{
 		log:               iOpts.Logger(),
 		metrics:           newWriterMetrics(scope),
+		rand:              rand.New(rand.NewSource(rand.Int63())),
```
Comment (xichen2020) [inline backend/msgpack/writer.go]: nit: perhaps just use `rand.NewSource(time.Now().UnixNano())`? not sure if the first number generated by `rand.Int63` will always be the same value.

**Pull request merged**